bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamIT.java

```diff
@@ -199,67 +175,74 @@ private void testWriteToBigtable() {
+    properties.applyTo(options);
     options.setJobName("testReadFromBigtable-" + System.currentTimeMillis());
     LOG.info(
         String.format("Started readFromBigtable test with jobName as: %s", options.getJobName()));
 
     Scan scan = new Scan();
     scan.setFilter(new FirstKeyOnlyFilter());
 
-    CloudBigtableScanConfiguration config =
+    CloudBigtableScanConfiguration.Builder configBuilder =
         new CloudBigtableScanConfiguration.Builder()
-            .withProjectId(projectId)
-            .withInstanceId(instanceId)
-            .withTableId(TABLE_NAME.getNameAsString())
-            .withScan(scan)
-            .withConfiguration(BIGTABLE_ADMIN_HOST_KEY, adminEndpoint)
-            .withConfiguration(BIGTABLE_HOST_KEY, dataEndpoint)
-            .build();
+            .withProjectId(properties.getProjectId())
+            .withInstanceId(properties.getInstanceId())
+            .withTableId(tableName.getNameAsString())
+            .withScan(scan);
+
+    properties
+        .getDataEndpoint()
+        .ifPresent(endpoint -> configBuilder.withConfiguration(BIGTABLE_HOST_KEY, endpoint));
+    properties
+        .getAdminEndpoint()
+        .ifPresent(endpoint -> configBuilder.withConfiguration(BIGTABLE_ADMIN_HOST_KEY, endpoint));
+
+    CloudBigtableScanConfiguration config = configBuilder.build();
 
     Pipeline pipeLine = Pipeline.create(options);
     PCollection<Long> count =
         pipeLine
             .apply("Read from BT", Read.from(CloudBigtableIO.read(config)))
-            .apply("Count", Count.<Result>globally());
+            .apply("Count", Count.globally());
 
     PAssert.thatSingleton(count).isEqualTo(TOTAL_ROW_COUNT);
-    return pipeLine;
-  }
 
-  @Test
-  public void testRunner() {
-    try {
-      // Submitted write pipeline to mutate the Bigtable.
-      testWriteToBigtable();
-
-      Pipeline result = testReadFromBigtable();
-      PipelineResult.State readJobStatue = result.run().waitUntilFinish();
-
-      Assert.assertEquals(PipelineResult.State.DONE, readJobStatue);
-    } catch (Exception ex) {
-      ex.printStackTrace();
-      throw new AssertionError("Exception occurred while pipeline execution");
-    }
+    PipelineResult.State result = pipeLine.run().waitUntilFinish();
+    Assert.assertEquals(PipelineResult.State.DONE, result);
   }
 
   private static byte[] createRandomValue() {
     byte[] bytes = new byte[CELL_SIZE];
     new Random().nextBytes(bytes);
     return bytes;
   }
-
-  private DataflowPipelineOptions createOptions() {
-    DataflowPipelineOptions options = PipelineOptionsFactory.as(DataflowPipelineOptions.class);
-    options.setProject(projectId);
-    options.setRegion(region);
-    options.setStagingLocation(stagingLocation + "/stage");
-    options.setTempLocation(stagingLocation + "/temp");
-    options.setRunner(DataflowRunner.class);
-    options.setWorkerMachineType(workerMachineType);
```

Comment:
Again, we are loosing this option, flagging to make sure we are making a conscious decision.

Reply:
yep this was intentional